src/Catty/ViewController/CatrobatTableViewController/CatrobatTableViewController.m

```diff
@@ -290,17 +290,65 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
 - (void)configureImageCell:(UITableViewCell <CatrobatImageCell>*)cell atIndexPath:(NSIndexPath*)indexPath
 {
     cell.backgroundColor = UIColor.background;
-    cell.titleLabel.text = [self.cells objectAtIndex:indexPath.row];
+    if (indexPath.row == 0) {
+        UILabel *subtitleLabel = (UILabel*)[cell viewWithTag:kSubtitleLabelTag];
+        subtitleLabel.textColor = UIColor.textTint;
+        subtitleLabel.text = [self.cells objectAtIndex:indexPath.row];
+    } else {
+        cell.titleLabel.text = [self.cells objectAtIndex:indexPath.row];
+    }
+    
+    ProjectLoadingInfo *info = nil;
+    NSArray* projectLoadingInfos = [Project allProjectLoadingInfos];
+    for (ProjectLoadingInfo *projectLoadingInfo in projectLoadingInfos) {
+        if ([projectLoadingInfo.visibleName isEqualToString:self.lastUsedProject.header.programName])
+            info = projectLoadingInfo;
+    }
+
+    if (indexPath.row == 0) {
+        
+        NSArray *fallbackPaths = @[[[NSString alloc] initWithFormat:@"%@%@", info.basePath, kScreenshotFilename],
```

Comment:
Why do we need those fallback paths here as well? Shouldn't that all be loaded from the `loadPreviewImageAndCache` method?

Reply:
fallbackPaths is needed there because if the `imageCache` is already having the image cached for any paths, then it can directly load and return. If not, only then we call `loadPreviewImageAndCache` method passing an instance of `ProjectLoadingInfo`. So `loadPreviewImageAndCache` won't get called every time. I hope that's what you meant.

Reply:
I think this is not ideal because now similar logic is at two places. How about integrating the cache to our new method as well?